addons/sale_loyalty/wizard/sale_loyalty_reward_wizard.py

```diff
@@ -52,3 +52,17 @@ def action_apply(self):
             raise ValidationError(_('Coupon not found while trying to add the following reward: %s', self.selected_reward_id.description))
         self.order_id._apply_program_reward(self.selected_reward_id, coupon, product=self.selected_product_id)
         return True
+
+    @api.model
+    def get_reward_info(self, order_id):
+        self.order_id = order_id
+        order = self.env['sale.order'].browse(order_id)
+        coupon_rewards = order._get_claimable_rewards()
+        reward_info = []
+        for coupon, rewards in coupon_rewards.items():
+            for reward in rewards:
+                points = order._get_real_points_for_coupon(coupon)
+                cost = reward.required_points
+                [(reward_id, name)] = reward.name_get()
+                reward_info.append([reward_id, name, points, cost, coupon.id])
```

Comment:
By default `display_name` computes from the name_get, using it will also batch the compute across all the rewards in our pool of rewards.
```suggestion
                reward_info.append([reward.id, reward.display_name, points, cost, coupon.id])
```